tests/Test/Xrefcheck/URIParsingSpec.hs

```diff
@@ -0,0 +1,72 @@
+{- SPDX-FileCopyrightText: 2022 Serokell <https://serokell.io>
+ -
+ - SPDX-License-Identifier: MPL-2.0
+ -}
+
+{-# LANGUAGE QuasiQuotes #-}
+
+module Test.Xrefcheck.URIParsingSpec where
+
+import Universum
+
+import Test.Hspec (Spec, describe, it)
+import Test.HUnit (assertBool)
+import Text.URI (URI)
+import Text.URI.QQ (uri)
+
+import Xrefcheck.Verify (parseUri, VerifyError (..))
+
+spec :: Spec
+spec = do
+  describe "URI parsing should be successful" $ do
+    it "Without the special characters in the query strings" do
+      parsed <- traverse parseUri' withoutSpecChars
+      assertBool "Failed to parse the link"
+        $ parsed == withoutSpecCharsExpected
```

Comment:
Instead of using `assertBool msg $ x == y`, let's use ``x `shouldBe` y``, as it gives us much better error messages (i.e. `assertBool` just prints a generic error message, `shouldBe` prints the expected and the actual values as well)

Reply:
In fact, I think these tests could be written more succinctly (and with less indirection, and more readable) like this:

```hs
    it "Without the special characters in the query strings" do
      parseUri' "https://example.com/?q=a&p=b#fragment" `shouldReturn`
        Right [uri|https://example.com/?q=a&p=b#fragment|]

      parseUri' "https://example.com/path/to/smth?q=a&p=b" `shouldReturn`
        Right [uri|https://example.com/path/to/smth?q=a&p=b|]
```